modules/cloudflare_delegation/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "cloudflare_zone" "this" {
```

Comment:
not sure it's ok. because this will create the zone in cloudflare no ? you should use a "data" to retrieve the cloudflare zone-id

Reply:
so if you create a "test.tamedia.ch" route53 zone in AWS, you'll need to create the "NS" record in the existing cloudflare "tamedia.ch" zone.
with this code, you'll create a "test.tamedia.ch" zone in cloudflare